### PR TITLE
[IMP] mrp_final_product_lot: Improve module behaviour

### PR DESCRIPTION
**Pull request by maurochip**
- Added help string in field **manual_production_lot**
- Refactored readonly to use states
- Make **lot_id** in **mrp.product.produce** no longer required (as it will be
  automatically assigned)
- Refactored the function **action_produce** to fix some bugs

### CODE REVIEW

File: mrp_final_product_lot/models/mrp_production.py [hunk 1]
```diff
@@ -38,8 +42,7 @@ def action_produce(
                         code = '-'.join(lots.mapped('name'))
                     vals = {'name': code,
                             'product_id': production.product_id.id}
-                    new_lot = lot_obj.create(vals)
-                    vals = {'lot_id': new_lot.id}
-                    wiz.write(vals)
+                    lot = lot_obj.create(vals)
+                wiz.lot_id = lot
```
Comment (pedrobaeza) [inline mrp_final_product_lot/models/mrp_production.py:46]: Shouldn't this be `wiz.lot_id = lot.id`?

File: mrp_final_product_lot/views/mrp_production_view.xml [hunk 2]
```diff
@@ -8,15 +9,23 @@
...
             </field>
         </record>
+
+        <record id="mrp_product_produce_form_view_inh_notlotrequired" model="ir.ui.view">
+            <field name="name">mrp.product.produce.form.view.inh.notlotrequired</field>
+            <field name="model">mrp.product.produce</field>
+            <field name="inherit_id" ref="mrp.view_mrp_product_produce_wizard"/>
+            <field name="arch" type="xml">
+                <field name="lot_id" position="attributes">
+                    <attribute name="attrs"/>
```
Comment (pedrobaeza) [inline mrp_final_product_lot/views/mrp_production_view.xml:25]: For what is this needed?

**Comment by maurochip** [comment]
If the product has checked lot traceabilty, the field **lot_id** in the produce wizard will be required. It will not allow me to create automatically the lot, for this reason I need that view.
`wiz.lot_id = lot` is correctly, I assign a object to object. I'm not sure, but I think `wiz.lot_id = lot.id` doesn't work.
`wiz.lot_id = lot` is the same as doing `wiz.write({'lot_id': lot.id})`.

**Comment by pedrobaeza** [comment]
`wiz.lot_id = lot.id` is for sure allowed. I remember some cases where the other is not always allowed (maybe on direct write expressions), but if you say that it works, then :+1:

**Pull request merged**